### DISCUSSION TITLE
contentlayer alternatives

### DISCUSSION THREAD

**Discussion by suntudo** [contentlayerdev/contentlayer#628] [General]
I'm a big fan of contentLayer but am starting to look into migrating to something else given the uncertainty around whether it will continue to be maintained. Has any one else started looking around to see what might be a good alternative? I'm specifically looking at libraries for Next14 and beyond.

[MDXTS](https://www.mdxts.dev/) looks promising, but it's still in alpha.

**Comment by airtonix** [comment]
https://github.com/contentlayerdev/contentlayer/issues/429

**Comment by fraineralex** [comment]
Hi @suntudo

It's understandable to explore alternatives, especially with considerations about ongoing maintenance. Here are a few alternatives you might find interesting:

1. [Velite](https://github.com/zce/velite): Converts Markdown, MDX, YAML, JSON, and more into your app's data layer with Zod schema.

2. [mdx-content](https://www.npmjs.com/package/mdx-content): A fully typed MDX loader for your project.

3. [typed-mdx](https://github.com/BearStudio/typed-mdx): Offers validation and typing for MDX Frontmatter.

As you mentioned, [MDXTS](https://www.mdxts.dev/) is also a promising option, albeit still in alpha. It's great to explore different alternatives and see what fits your project's needs. 

Good luck with your exploration!

**Comment by marekzan** [comment]
There is also https://www.content-collections.dev/

**Comment by timlrx** [comment]
Updated fork - [Contentlayer2](https://github.com/timlrx/contentlayer2)

**Comment by airtonix** [comment]
https://github.com/datopian/markdowndb

**Comment by cpatti97100** [comment]
https://github.com/contentlayerdev/contentlayer/issues/651
